### PR TITLE
poke: workspace segementation

### PR DESCRIPTION
**Pull request by spolu**
Fixes: https://github.com/dust-tt/tasks/issues/264

The segmentation field will be visible client-side to user that would want to look for it. But I think that passes the front test page no problem.

### CODE REVIEW

File: front/pages/poke/[wId]/index.tsx [hunk 1]
```diff
@@ -270,6 +276,29 @@ const WorkspacePage = ({
     }
   );
 
+  const { submit: onWorkspaceUpdate } = useSubmitFunction(
+    async (segmentation: WorkspaceSegmentationType) => {
+      try {
+        const r = await fetch(`/api/poke/workspaces/${owner.sId}`, {
+          method: "POST",
```
Comment (PopDaph) [inline front/pages/poke/[wId]/index.tsx:283]: PATCH?

File: front/pages/api/poke/workspaces/[wId]/index.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `The request body is invalid: ${pathError}`,
+          },
+        });
+      }
+      const body = bodyValidation.right;
+      const workspace = await Workspace.findOne({
```
Comment (PopDaph) [inline front/pages/api/poke/workspaces/[wId]/index.ts:56]: No queries on API routes does not apply to Poké?

File: front/lib/auth.ts [hunk 3]
```diff
@@ -292,6 +292,7 @@ export class Authenticator {
           name: this._workspace.name,
           allowedDomain: this._workspace.allowedDomain || null,
           role: this._role,
+          segmentation: this._workspace.segmentation || null,
```
Comment (PopDaph) [inline front/lib/auth.ts:295]: We could do: 
```
segmentation: this.isDustSuperUser() ? this._workspace.segmentation : null,
```

**Review by PopDaph — DISMISSED** [review]
Approving cause it was probably already discussed! 🙏🏻 

If it was me doing it I would have challenged adding a string over a JSONB field, because I'm expecting the interesting/null to not be sufficient if we want to do segmentation. I'm sure we'll want to store more internal workspace data at some point, such as number of employees or country or whatever. (Would be scrappy but we already have a tool on Poké to display and edit jsons for the Subscription)

File: front/pages/api/poke/workspaces/[wId]/index.ts (hunk 2 above)
Comment (spolu) [reply front/pages/api/poke/workspaces/[wId]/index.ts:56]: Well it's a super internal thing and I think it does not deserve a lib/api logic as such.

 I removed a bunch of queries from poke routes this morning to be fair :) https://github.com/dust-tt/dust/pull/2752

File: front/pages/poke/[wId]/index.tsx (hunk 1 above)
Comment (spolu) [reply front/pages/poke/[wId]/index.tsx:283]: sure

File: front/lib/auth.ts (hunk 3 above)
Comment (spolu) [reply front/lib/auth.ts]: Actually the goal is maybe to use it in the product at some point so I'll keep it everywhere

**Comment by spolu** [comment]
Deploy: 
- deploy `front-edge` + run init_db
- deploy `front`

File: front/pages/api/poke/workspaces/[wId]/index.ts (hunk 2 above)
Comment (spolu) [reply front/pages/api/poke/workspaces/[wId]/index.ts]: Moved it to an api function :)

**Pull request merged**